lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
`test_tcp_options` was testing the results of `Options.marshal`, but not the presence of the options field in a generated packet.  I've added a test which fails on current master in https://github.com/yomimono/mirage-tcpip/commit/abbdbe9c5ad1dbb161143981f9a4534d2c2cbec1 .